### ISSUE TITLE
Bug(website): when `ts` hash parameter is missing, nonsense content flashes in editor

### ISSUE THREAD

**Issue by abrahamguo** [typescript-eslint/typescript-eslint#8780]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

website

### Playground Link

https://typescript-eslint.io/play/#code=MYewdgziA2CmB00QHMAUByAEraSAEA7iAE7QAmAhOgJRA

### Repro Code

```TypeScript
// Go to the URL above
```

### Expected Result

The `code` tab should show `console.log('Hello world!')`.

### Actual Result

For a brief moment, the `code` tab flashes with `ààà`.

### Additional Info

The bug only appears if the `ts` hash parameter is missing. If `ts=5.4.3` is added to the hash, the issue does not occur.

### Versions

Latest live website

**Comment by Josh-Cena** [comment]
Curiously you can reproduce the same behavior in the TS playground: https://www.typescriptlang.org/play/#code=MYewdgziA2CmB00QHMAUByAEraSAEA7iAE7QAmAhOgJRA

These characters come from `sandboxInstance.getWorkerProcess()`: https://github.com/typescript-eslint/typescript-eslint/blob/2619c3b5e6718a6b70519d9e9807ee7acb745369/packages/website/src/components/editor/useSandboxServices.ts#L84

And is likely a bug or feature of the underlying Monaco sandbox. I'm not sure how we can help here.

**Comment by Josh-Cena** [comment]
The problem is more curious than `ts` hash parameter is missing. The problem is that `code=` cannot appear at the start of the hash.

- This flashes: https://typescript-eslint.io/play/#code=MYewdgziA2CmB00QHMAUByAEraSAEA7iAE7QAmAhOgJRA&ts=5.4.3
- This does not: https://typescript-eslint.io/play/#&code=MYewdgziA2CmB00QHMAUByAEraSAEA7iAE7QAmAhOgJRA
- Nor this: https://typescript-eslint.io/play/#fileType=.ts&code=MYewdgziA2CmB00QHMAUByAEraSAEA7iAE7QAmAhOgJRA

**Comment by Josh-Cena** [comment]
Here's one observation: https://www.typescriptlang.org/play/#code=MYewdgziA2CmB00QHMAUByAEraSAEA7iAE7QAmAhOgJRA eventually becomes https://www.typescriptlang.org/play/?#code/Aco. And indeed `#code=Aco` is the encoding of `ààà`. I'm not sure what `Aco` is supposed to represent. Maybe it's the last three characters of `Monaco`?

**Comment by Josh-Cena** [comment]
Ha, I've figured it out. Apparently when the sandbox runs `getInitialCode`: https://github.com/microsoft/TypeScript-Website/blob/57e19fd99f8128f9abd3810a89b9856f1b36b32b/packages/sandbox/src/getInitialCode.ts#L8 It tests using `if (location.hash.startsWith("#code"))` but then parses the code from the URL using `location.hash.replace("#code/", "").trim()`. If the hash starts with `#code=`, then it goes into the branch but fails to extract the correct lzstring. And indeed, `lzstring.decompressFromEncodedURIComponent(
"#code=MYewdgziA2CmB00QHMAUByAEraSAEA7iAE7QAmAhOgJRA")` would give you `"ààà"`.

The correct solution here may be to disable default text fetching, because we later do our own state parsing from the URL anyway. This removes the capability to read from local storage though. Meanwhile I've sent https://github.com/microsoft/TypeScript-Website/pull/3523.

**Comment by Josh-Cena** [comment]
I believe the TS-website PR is sufficient to solve this problem.
